optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/config/PlannerBenchmarkConfig.java

```diff
@@ -800,27 +799,9 @@ protected void generateSolverBenchmarkConfigNames() {
 
     protected int resolveParallelBenchmarkCount() {
         int availableProcessorCount = Runtime.getRuntime().availableProcessors();
-        int resolvedParallelBenchmarkCount;
-        if (parallelBenchmarkCount == null) {
-            resolvedParallelBenchmarkCount = 1;
-        } else if (parallelBenchmarkCount.equals(PARALLEL_BENCHMARK_COUNT_AUTO)) {
-            resolvedParallelBenchmarkCount = resolveParallelBenchmarkCountAutomatically(availableProcessorCount);
-        } else {
-            resolvedParallelBenchmarkCount = ConfigUtils.resolveThreadPoolSizeScript(
-                    "parallelBenchmarkCount", parallelBenchmarkCount, PARALLEL_BENCHMARK_COUNT_AUTO);
-        }
-        if (resolvedParallelBenchmarkCount < 1) {
-            throw new IllegalArgumentException("The parallelBenchmarkCount (" + parallelBenchmarkCount
-                    + ") resulted in a resolvedParallelBenchmarkCount (" + resolvedParallelBenchmarkCount
-                    + ") that is lower than 1.");
-        }
-        if (resolvedParallelBenchmarkCount > availableProcessorCount) {
-            logger.warn("Because the resolvedParallelBenchmarkCount ({}) is higher "
-                    + "than the availableProcessorCount ({}), it is reduced to "
-                    + "availableProcessorCount.", resolvedParallelBenchmarkCount, availableProcessorCount);
-            resolvedParallelBenchmarkCount = availableProcessorCount;
-        }
-        return resolvedParallelBenchmarkCount;
+        return ConfigUtils.resolveThreadPoolSize("parallelBenchmarkCount", parallelBenchmarkCount,
+                () -> resolveParallelBenchmarkCountAutomatically(availableProcessorCount),
+                () -> availableProcessorCount, logger);
```

Comment:
If PARALLEL_BENCHMARK_COUNT_AUTO isn't used any more, it should be removed too.

But that's still used in javadocs, and replacing it with ConfigUtils#THREAD_COUNT_AUTO makes no sense, because not a THREAD_COUNT. One parallel benchmark or 1 solverManager.parallel may actually use multiple threads. So calling it threadCount is confusing. Also,  resolveThreadPoolSize() is a bad name I guess - **we can rename that to resolvePoolSize()?**

I see 2 options:
1) rename - ConfigUtils.THREAD_COUNT_AUTO to just AUTO. But then every resolveThreadPoolSize() call must have an AUTO option (which is currently the case, so I guess that's fine).
2) Don't generalize "auto", like it is before this PR (doesn't impact the other changes in this PR).